src/devices/machine/ibmps2.h

```diff
@@ -0,0 +1,265 @@
+
+	IRQ_CALLBACK_MEMBER(irq_ack_cb);
+
+protected:
+	required_device<cpu_device> m_maincpu;
+    required_device<ram_device> m_ram;
+	optional_device<nvram_device> m_nvram;
+	required_device<mca16_device> m_mcabus;
+	required_device<pic8259_device> m_pic8259_master;
+	required_device<pic8259_device> m_pic8259_slave;
+	required_device<ibm72x7377_device> m_dmac;
+	required_device<ps2_pit_device> m_pit;
+	required_device<speaker_sound_device> m_speaker;
+	required_device<mc146818_device> m_mc146818;
+	required_device<ps2_keyboard_controller_device> m_keybc;
+    required_device<pc_kbdc_device> m_ps2_con;
+	required_device<pc_kbdc_device> m_aux_con;
+	required_device<mca16_slot_device> m_planar_fdc;
+	required_device<mca16_slot_device> m_planar_uart;
+	required_device<mca16_slot_device> m_planar_lpt;
+	required_device<mca16_slot_device> m_planar_vga;
+
+	optional_device<mca16_slot_device> m_mcaslot_1;
+	optional_device<mca16_slot_device> m_mcaslot_2;
+	optional_device<mca16_slot_device> m_mcaslot_3;
+	optional_device<mca16_slot_device> m_mcaslot_4;
+	optional_device<mca16_slot_device> m_mcaslot_5;
+	optional_device<mca16_slot_device> m_mcaslot_6;
+	optional_device<mca16_slot_device> m_mcaslot_7;
+	optional_device<mca16_slot_device> m_mcaslot_8;
+
+	uint16_t m_sram_size;
+	uint16_t m_planar_id;
+
+	std::unique_ptr<uint8_t[]> m_sram;
+    uint16_t m_nvram_index = 0;
+
+	void ps2_32_map(address_map& map);
+	void ps2_32_io(address_map& map);
+
+	void ps2_16_map(address_map& map);
+	void ps2_16_io(address_map& map);
+
+	void device_start() override;
+	void device_reset() override;
+	virtual void device_add_mconfig(machine_config &config) override;
+    virtual void device_config_complete() override;
+
+	void add_mca16(machine_config &config);
+	void add_mca32(machine_config &config);
+
+	void speaker_set_spkrdata(uint8_t data);
+
+	uint8_t m_at_spkrdata = 0;
+	uint8_t m_pit_out2 = 0;
+	bool m_write_gate_2 = 0;
+	
+	uint8_t m_at_speaker = 0;
+	uint8_t m_channel_check = 0;
+	uint8_t m_nmi_enabled = 0;
+	uint8_t m_parity_check_enabled = 0;
+	uint8_t m_channel_check_enabled = 0;
+	uint8_t m_memory_control = 0;
+	bool m_gate_a20 = 0;
+	bool m_nmi_flag = 0;
+	bool m_alt_hot_reset = 0;
+	
+	DECLARE_WRITE_LINE_MEMBER(pit8254_out2_changed);
+	DECLARE_WRITE_LINE_MEMBER(irq0_w);
+	DECLARE_WRITE_LINE_MEMBER(watchdog_w);
+	DECLARE_WRITE_LINE_MEMBER(cd_sfdbk_w);
+	DECLARE_WRITE_LINE_MEMBER(gate_a20_w);
+
+	uint8_t get_slave_ack(offs_t offset);
+
+	// System Control Ports A and B
+	uint8_t	porta_r();
+	void	porta_w(uint8_t data);
+
+	virtual uint8_t portb_r();
+	virtual void portb_w(uint8_t data);
+
+	// MCA Programmable Option Select
+	virtual uint8_t planar_pos_r(offs_t offset);
+	virtual void 	planar_pos_w(offs_t offset, uint8_t data);
+
+	uint8_t card_select_feedback_r(offs_t offset);
+	void update_system_board_io(uint8_t data);
+
+	uint8_t m_system_board_setup = 0;
+    uint8_t m_system_board_io_enable = 0;
+
+    uint8_t m_adapter_setup = 0;
+	bool m_cd_sfdbk = 0;
```

Comment:
Please don’t intermix data members and member functions – keep the data members together.